src/propagation/constant_cstring_propagator.cpp

```diff
@@ -1,12 +1,15 @@
+// clang-format: off
+// FIXME: These two includes have a dependent include order
```

Comment:
I think this might be occuring because `BlockVisitorManager<V>` uses `CStringVisitor`, which it assumes is defined at the time of template instantiation. Switching the include order borks this up.

I am unsure if this has wide impact because these are internal headers, I think the clean solution to this might be to have `block_visitor_manager.h` depend on `propagation_visitor.h` or merge those two altogether.

@ruoso @dbeer1 @azeemba , any comments?

Reply:
@dbeer1 Identified the issue. It was to do with function lookup failing because the previous include order made it so that `operator<<(ostream&, const PropagationResult<T>&)` had a visible definition prior to its call-site in `ConstantPropagator<T>`.

Upon switching the include order, this was no longer the case. We move the definition to the namespace `clangmetatool::propoagation` as the compiler note suggests. This should build with both GCC/clang now.